lib/launcher.rb

```diff
@@ -119,18 +119,18 @@ def webrick_server(app)
     end
 
     def launch
-      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
-
-      http_app = http_app()
-      https_app = https_app()
-      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
-
       if SETTINGS.daemon
         check_pid
         Process.daemon
         write_pid
       end
 
+      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
+
+      http_app = http_app()
+      https_app = https_app()
+      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
```

Comment:
I don't think it checks if modules are present - only that `SETTINGS.http_port` is given for HTTP, or that `SETINGS.ssl_private_key` etc are given for HTTPS (curiously, not https_port).  The apps will still be returned even if no modules are configured, I think.